addons/stock/static/src/js/report_stock_forecasted.js

```diff
@@ -323,6 +324,23 @@ const ReplenishReport = clientAction.extend({
         }).then((result) => {
             return this._reloadReport();
         });
+    },
+
+    /**
+     * Reserve the specified model/id, then reload this report.
+     *
+     * @returns {Promise}
+     */
+    _onClickReserve: function(ev) {
+        const model = ev.target.getAttribute('model');
+        const modelId = parseInt(ev.target.getAttribute('model-id'));
+        this._rpc( {
+            model: model,
+            args: [[modelId]],
+            method: 'action_assign'
+        }).then((result) => {
+            return this._reloadReport();
+        });
```

Comment:
```suggestion
        this._rpc({
            model,
            args: [[modelId]],
            method: 'action_assign'
        }).then(() => this._reloadReport());
```
Also, should return `this._rpc` or remove the `@returns {Promise}` in the docstring.